### PR TITLE
(BOLT-359) Fixes for the facts module

### PR DESCRIPTION
**Pull request by MikaelSmith**
Adds tasks to the gem, adds fallback to kernel for bash facts (which adds support for macOS), and adds integration tests for Windows and Linux.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Yeah, we probably do want acceptance tests. I could move the integration tests there.

File: modules/facts/tasks/powershell.ps1 [hunk 1]
```diff
@@ -3,17 +3,20 @@
 # Delegate to facter if available
 if (Get-Command facter -ErrorAction SilentlyContinue) {
     facter --json
-} else {
-    if ([System.Environment]::OSVersion.Platform -gt 2) { # [System.PlatformID]::Win32NT
-@'
+}
+else {
+    if ([System.Environment]::OSVersion.Platform -gt 2) {
+        # [System.PlatformID]::Win32NT
```
Comment (mruzicka) [inline modules/facts/tasks/powershell.ps1]: This comment refers to the value "2" on the previous line so should be on that line. It is the value of the `[System.PlatformID]::Win32NT` constant. For some reason using that constant doesn't work on Windows Server Core but it does work on other versions of Windows.

Comment (MikaelSmith) [reply modules/facts/tasks/powershell.ps1]: This was mostly auto-reformatting by the Powershell extensions for Visual Studio Code. We can revert it. The only actual change was to the error kind.

Comment (mruzicka) [reply modules/facts/tasks/powershell.ps1]: No need to revert it, I'll put a PR up moving the comment before the condition and making it clearer.

**Pull request merged**